packages/cli/src/commands/locations/modes/delete.ts

```diff
@@ -0,0 +1,34 @@
+import { Flags } from '@oclif/core'
+import { APICommand } from '@smartthings/cli-lib'
+import { chooseMode } from '../../../lib/commands/locations/modes-util'
+
+
+export default class ModesDeleteCommand extends APICommand<typeof ModesDeleteCommand.flags> {
+	static description = 'delete a mode'
+
+	static flags = {
+		...APICommand.flags,
+		location: Flags.string({
+			char: 'l',
+			description: 'a specific location to query',
+			helpValue: '<UUID>',
+		}),
+	}
+
+	static args = [{
+		name: 'id',
+		description: 'mode UUID',
+	}]
+
+	static examples = [
+		{ description: 'select a mode from a list of all modes and delete it', command: 'smartthings locations:modes:delete' },
+		{ description: 'select a mode from a list of modes in a specified location and delete it', command: 'smartthings locations:modes:delete --location=5dfd6626-ab1d-42da-bb76-90def3153998' },
+		{ description: 'delete a specified mode', command: 'smartthings locations:modes:delete 636169e4-8b9f-4438-a941-953b0d617231' },
+	]
+
+	async run(): Promise<void> {
+		const [modeId, locationId] = await chooseMode(this, this.flags.location, this.args.id)
```

Comment:
I've put the call to `chooseLocation` at the beginning of `chooseMode` because I think we'll always want to call it at that point and it seemed cleaner than other options.  What do you think?

Reply:
I wouldn't have thought of that but it seems like a good idea to me.